packages/share/ios/Classes/FLTSharePlugin.m

```diff
@@ -57,8 +98,19 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
                                   binaryMessenger:registrar.messenger];
 
   [shareChannel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
+    NSDictionary *arguments = [call arguments];
+    NSNumber *originX = arguments[@"originX"];
+    NSNumber *originY = arguments[@"originY"];
+    NSNumber *originWidth = arguments[@"originWidth"];
+    NSNumber *originHeight = arguments[@"originHeight"];
+
+    CGRect originRect = CGRectZero;
+    if (originX != nil && originY != nil && originWidth != nil && originHeight != nil) {
```

Comment:
Nice :). Just to ensure, objective-c is not like JavaScript in this place? in Javascript `originX = 0` would fail the `if (originX)` check as 0 resolves to false. I hope in objective-c that is not he case?

Reply:
It seems obj-C behaves the same as Javascript: https://stackoverflow.com/a/6177766

Reply:
But in this case, it seems `originWhatevers` are Strings that get [cast to a double](https://developer.apple.com/documentation/foundation/nsstring/1414031-doublevalue?language=objc) on line 109?

`[originX doubleValue]`...

Reply:
[NSNumber seems to cast to false when it's `0`](https://developer.apple.com/documentation/foundation/nsnumber), I'd leave the `!= nil` check in this case too :P

Reply:
Thx for checking this :). I was not sure either and could not find a clear answer. Tested around a bit and think we are save without the nil check.